docs/serving/samples/hello-world/helloworld-php/index.md

```diff
@@ -5,4 +5,209 @@ weight: 1
 type: "docs"
 ---
 
-{{% readfile file="README.md" %}}
+A simple web app written in PHP that you can use for testing. It reads in an env
+variable `TARGET` and prints `Hello ${TARGET}!`. If `TARGET` is not specified,
+it will use `World` as the `TARGET`.
+
+Follow the steps below to create the sample code and then deploy the app to your
+cluster. You can also download a working copy of the sample, by running the
+following commands:
+
+```shell
+git clone -b "{{< branch >}}" https://github.com/knative/docs knative-docs
+cd knative-docs/docs/serving/samples/hello-world/helloworld-php
+```
+
+## Before you begin
+
+- A Kubernetes cluster with Knative installed and DNS configured. Follow the
+  [installation instructions](../../../../install/README.md) if you need to
+  create one.
+- [Docker](https://www.docker.com) installed and running on your local machine,
+  and a Docker Hub account configured (we'll use it for a container registry).
+
+## Recreating the sample code
+
+1. Create a new directory and cd into it:
+
+   ```shell
+   mkdir app
+   cd app
+   ```
+
+1. Create a file named `index.php` and copy the code block below into it:
+
+   ```php
+   <?php
+   $target = getenv('TARGET', true) ?: 'World';
+   echo sprintf("Hello %s!", $target);
+   ```
+
+1. Create a file named `Dockerfile` and copy the code block below into it. See
+   [official PHP docker image](https://hub.docker.com/_/php/) for more details.
+
+   ```docker
+   # Use the official PHP 7.3 image.
+   # https://hub.docker.com/_/php
+   FROM php:7.3-apache
+
+   # Copy local code to the container image.
+   COPY index.php /var/www/html/
+
+   # Use the PORT environment variable in Apache configuration files.
+   RUN sed -i 's/80/${PORT}/g' /etc/apache2/sites-available/000-default.conf /etc/apache2/ports.conf
+
+   # Configure PHP for development.
+   # Switch to the production php.ini for production operations.
+   # RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
+   # https://hub.docker.com/_/php#configuration
+   RUN mv "$PHP_INI_DIR/php.ini-development" "$PHP_INI_DIR/php.ini"
+   ```
+
+1. Create a `.dockerignore` file to ensure that any files related to a local
+   build do not affect the container that you build for deployment.
+
+   ```ignore
+   Dockerfile
+   README.md
+   vendor
```

Comment:
You really want to ignore the Dockerfile ? it needs to be part of the context tar that is send to the Docker daemon. Also `vendor` in the context of a PHP sample app doesn't make any sense.

Reply:
tbh, I think you could just skip this step. Its not necessary, add only a little overhead (the README) and is of little value.